src/components/Table.jsx

```diff
@@ -19,13 +26,28 @@ function Table({ team, searchTerm, onSearchChange }) {
           </tr>
         </thead>
         {
-          team.filter((item) => {
-            if (item === true) {
-              return true;
-            } if (item.name.toLowerCase().includes(searchTerm.toLowerCase())) {
+          data.filter((item) => {
+            const salaryRange = () => {
+              if (minValue === '' && item.salary <= maxValue) return true;
+              if (maxValue === '' && item.salary >= minValue) return true;
+              if (item.salary >= minValue && item.salary <= maxValue) return true;
+              return false;
+            };
+
+            const searchBar = () => {
+              if (item.name.toLowerCase().includes(searchTerm.toLowerCase())) {
+                return true;
+              }
+              return false;
+            };
+
+            const typeButton = () => {
+              if (team === 'Employer') return item.type === 'Employer';
+              if (team === 'Employee') return item.type === 'Employee';
               return true;
-            }
-            return false;
+            };
+
+            return salaryRange() && searchBar() && typeButton();
```

Comment:
It would be better if you make these functions pure. I mean they will not use variables outside of its own scope. Pass filter variables as func arguments